src/Sentry.Unity.Editor/AutoInstrumentation/PerformanceAutoInstrumentation.cs

```diff
@@ -0,0 +1,217 @@
+                        return;
+                    }
+
+                    logger?.LogInfo("Compilation of '{0}' finished. Running Performance Auto Instrumentation.", assemblyPath);
+
+                    var stopwatch = new Stopwatch();
+                    stopwatch.Start();
+
+                    try
+                    {
+                        ModifyPlayerAssembly(logger, assemblyPath, sentryUnityAssemblyPath);
+                    }
+                    catch (Exception e)
+                    {
+                        logger?.LogError("Failed to add the performance auto instrumentation. " +
+                                         "The assembly has not been modified.", e);
+                    }
+
+                    stopwatch.Stop();
+                    logger?.LogInfo("Auto Instrumentation finished in '{0}'.", stopwatch.Elapsed);
+                }
+            };
+        }
+
+        private static TypeDefinition GetTypeDefinition(ModuleDefinition module, Type type)
+        {
+            var typeDefinition = module.GetType(type.FullName);
+            if (typeDefinition is null)
+            {
+                throw new Exception($"Failed to get type '{type.FullName}' from module '{module.Name}'");
+            }
+
+            return typeDefinition;
+        }
+
+        private static MethodReference ImportReference(ModuleDefinition module, MethodDefinition method)
+        {
+            var reference = module.ImportReference(method);
+            if (reference is null)
+            {
+                throw new Exception($"Failed to import '{method.FullName}' into '{module.Name}");
+            }
+
+            return reference;
+        }
+
+        private static MethodDefinition GetMethodDefinition(TypeDefinition typeDefinition, string name, Type[]? requiredParameters = null)
+        {
+            foreach (var method in typeDefinition.Methods)
+            {
+                if (method?.Name == name)
+                {
+                    switch (requiredParameters)
+                    {
+                        case null when !method.HasParameters:
+                            return method;
+                        case null:
+                            continue;
+                    }
+
+                    if (method.Parameters.Count != requiredParameters.Length)
+                    {
+                        continue;
+                    }
+
+                    // We go over all the required parameters and compare them to the parameters found in the method.
+                    var hasMatchingParameters = true;
+                    foreach (var parameter in requiredParameters)
+                    {
+                        var parameterDefinitions = method.Parameters.Where(p =>
+                            string.Equals(
+                                p.ParameterType.FullName,
+                                parameter.FullName,
+                                StringComparison.CurrentCulture))
```

Comment:
Parameter names shouldn't use the current culture.  Probably you want `Ordinal` - which is the default for strings.